include/exadg/operators/inverse_mass_operator.h

```diff
@@ -60,14 +62,50 @@ class InverseMassOperator
     this->matrix_free = &matrix_free_in;
     dof_index         = dof_index_in;
     quad_index        = quad_index_in;
+
+    cellwise_inverse_mass_not_available = not(matrix_free->get_dof_handler(dof_index)
+                                                .get_triangulation()
+                                                .all_reference_cells_are_hyper_cube());
+
+    if(cellwise_inverse_mass_not_available)
+    {
+      initialize_inverse_mass_operator_with_block_jacobi();
+    }
+  }
+
+  void
+  initialize_inverse_mass_operator_with_block_jacobi()
```

Comment:
This looks like a public function. Why not just inserting the code implemented here in this function into the above if branch?

Reply:
> Why not just inserting the code implemented here in this function into the above if branch?

I didn't do this to keep the code easier to read. Would it be a good solution to make this function private?

Reply:
Member functions that are not part of the public interface (used to interact with the object from the outside) should always be private/protected.

Reply:
made the function private in ad3d8b5b3b30e00d0a6242ef35c038ed8e4a8c2b.

Reply:
I do not see a need to have a separate private member function, but we do it in a similar way at other places in ExaDG, so its ok. (From a software design perspective, some people might argue that it is needless complexity to introduce a function here; others might argue that the code is structured in a better way, more readable ...).